plugins/ptp_operator/ptp_operator_plugin.go

```diff
@@ -314,3 +325,21 @@ func HasEqualInterface(a []*string, b []*ptp4lconf.PTPInterface) bool {
 	}
 	return true
 }
+
+// InitPubSubTypes ... initialize types of publishers for ptp operator
+func InitPubSubTypes() map[ptp.EventType]*ptpTypes.EventPublisherType {
+	publishers := make(map[ptp.EventType]*ptpTypes.EventPublisherType)
+	publishers[ptp.OsClockSyncStateChange] = &ptpTypes.EventPublisherType{
+		EventType: ptp.OsClockSyncStateChange,
+		Resource:  ptp.OsClockSyncState,
+	}
+	publishers[ptp.PtpClockClassChange] = &ptpTypes.EventPublisherType{
+		EventType: ptp.PtpClockClassChange,
+		Resource:  ptp.PtpClockClass,
+	}
+	publishers[ptp.PtpStateChange] = &ptpTypes.EventPublisherType{
+		EventType: ptp.PtpStateChange,
+		Resource:  ptp.PtpLockState,
+	}
```

Comment:
There are a few other event Types defined in the ptp event types file do we need to add the publishers for those eventTypes too?
https://github.com/redhat-cne/sdk-go/blob/main/pkg/event/ptp/types.go

Reply:
Not yet, right now we only support 3 types of events that are implemented. until we don't know what is needed, we are not adding it. The list is from API spec document